tests/cases/compiler/narrowingTypeofUndefined.ts

```diff
@@ -13,3 +15,8 @@ if (typeof a.error !== 'undefined') {
 else {
     a.result.prop; // number
 }
+
+// extra parens on purpose
+if (typeof (a.error) === 'undefined') {
+    a.result.prop; // number
+}
```

Comment:
This reported an error before this PR: https://github.com/microsoft/TypeScript/pull/51720

The issue was that `typeOfExpr.expression` was passed to `getDiscriminantPropertyAccess` here:
https://github.com/microsoft/TypeScript/pull/50344/files#diff-d9ab6589e714c71e657f601cf30ff51dfc607fc98419bf72e04f6b0fa92cc4b8R25331

and it got fixed by @ahejlsberg since he changed this and now `target` is passed there:
https://github.com/microsoft/TypeScript/pull/51720/files#diff-d9ab6589e714c71e657f601cf30ff51dfc607fc98419bf72e04f6b0fa92cc4b8R26520

This works since `target` is the result of `getReferenceCandidate`:
https://github.com/microsoft/TypeScript/pull/50344/files#diff-d9ab6589e714c71e657f601cf30ff51dfc607fc98419bf72e04f6b0fa92cc4b8R25329
and that includes "skip parenthesis"-like logic:
https://github.dev/microsoft/TypeScript/blob/c124d0e2604bac1a428c24b98a1dd6508d860812/src/compiler/checker.ts#L25182-L25185

Reply:
https://github.com/microsoft/TypeScript/pull/51720 has a test with optional chaining and parenthesis - that has worked before your PR despite the fact that the version without parenthesis reported an error. This test is kinda about the opposite situation - the version without parenthesis worked alright while the version with them reported an error.

I've concluded that they are slightly different and thus I've created a PR (I already had it locally as I was looking into fixing https://github.com/microsoft/TypeScript/issues/51700 when you picked it up). I understand that they can be seen as similar enough though. Feel free to merge or close - whatever you think is best here.